### DISCUSSION TITLE
TypeError: Cannot destructure property 'data' of '(intermediate value)' as it is undefined.

### DISCUSSION THREAD

**Discussion by TamaraJoniec** [vercel/next.js#58175] [Help] [answered]
### Summary

Hi guys,

I've agreed to help someone make a little tweak to their code, but I can't even get the website to load on localhost as it's showing this error: TypeError: Cannot destructure property 'data' of '(intermediate value)' as it is undefined.

I'm new at nextjs and coding in general. 
Things I've tried: npm install , checking node versions (they seem ok, in package.lock.json it says "engines": {
  "node": ">=6.0.0" 
}. 
I've looked online for similar examples of this issue but I can't work it out.

Please help!

### Additional information

```js
1 of 1 unhandled error
Server Error
TypeError: Cannot destructure property 'data' of '(intermediate value)' as it is undefined.

This error happened while generating the page. Any console logs will be displayed in the terminal window.
Source
lib/api.js (19:10) @ getPageData

  17 | 
  18 | export const getPageData = async slug => {
> 19 | const { data } = await client.query({
     |        ^
  20 |   query: PageDataQuery,
  21 |   variables: {
  22 |     id: slug,
Call Stack
process.processTicksAndRejections
node:internal/process/task_queues (95:5)
async getStaticProps
pages/index.tsx (19:15)
async Object.renderToHTML
file:///Users/tamarajoniec/blush-website/node_modules/next/dist/server/render.js (386:20)
async doRender
file:///Users/tamarajoniec/blush-website/node_modules/next/dist/server/base-server.js (687:38)
async cacheEntry.responseCache.get.isManualRevalidate.isManualRevalidate
file:///Users/tamarajoniec/blush-website/node_modules/next/dist/server/base-server.js (796:28)
async
file:///Users/tamarajoniec/blush-website/node_modules/next/dist/server/response-cache/index.js (80:36)

The api.js page the error is coming from: 
import client from '../apollo-client'
import { PageDataQuery } from '../queries/pageDataQuery'
import { WorkPathsQuery } from '../queries/workPathsQuery'
import { JobPathsQuery } from '../queries/jobPathsQuery'

const isPageWithPageFields = slug => {
  const whitelist = ['/terms', '/privacy', '/cookies', '/job']

  return whitelist.some(item => slug.includes(item))
}

const isJobPage = slug => {
  const whitelist = ['/job']

  return whitelist.some(item => slug.includes(item))
}
//this is the part that's bugging
export const getPageData = async slug => {
  const { data } = await client.query({
    query: PageDataQuery,
    variables: {
      id: slug,
      acfHomeFields: slug === '/',
      acfPageFields: isPageWithPageFields(slug),
      acfAboutTemplate: slug.includes('/about'),
      acfDesignForGoodTemplate: slug.includes('/design-for-good'),
      acfWorkImages: slug.includes('/work'),
      acfContactTemplate: slug.includes('/contact'),
      acfJobFields: isJobPage(slug),
    },
  })

  return data
}

export const getWorkPaths = async () => {
  const { data } = await client.query({
    query: WorkPathsQuery(),
  })

  return data
}

export const getJobPaths = async () => {
  const { data } = await client.query({
    query: JobPathsQuery(),
  })

  return data
}
```

### Example

https://blushcreate.com/

**Comment by icyJoseph** [comment]
Do you have all env vars setup? Or rather, is Apollo correctly setup in your local dev machine? It can be API keys, API URLs, special headers, etc, double check those first.

**Reply by TamaraJoniec** [reply]
I'm not sure how to check that, but I found the Apollo setup: 

`import { ApolloClient, InMemoryCache } from '@apollo/client'

const API_URL = process.env.WORDPRESS_API_URL

const client = new ApolloClient({
  uri: API_URL,
  cache: new InMemoryCache(),
})

export default client
`

**Reply by TamaraJoniec** [reply]
The website works fine online, it just won't work on my local host :(

**Reply by icyJoseph** [reply, marked as answer]
Yeah, so you need this value: `WORDPRESS_API_URL` in a `.env` file in your local machine. Sometimes there's also a `APOLLO` API key that's needed.

**Reply by TamaraJoniec** [reply]
I've just added the wordpress api url to the env.local, its the first time I've done this so its taking a while to figure out. Now I've got the error index.js:594 Uncaught ApolloError: Response not successful: Received status code 404
    at new ApolloError (/Users/tamarajoniec/blush-website/node_modules/@apollo/client/errors/errors.cjs:34:28)

**Reply by icyJoseph** [reply]
I see, but that means that the API is not reachable, "not found". 

Maybe you can use something like Postman, to make sure you can run queries against that Wordpress API URL.

**Reply by Nishant-Vapta** [reply]
I am just starting off and configure parcel for my react app, but I dont have any .env file exist. 
![image](https://github.com/vercel/next.js/assets/152993379/01bb426d-f301-41cd-abf7-ecc0ddca2d67)

**Reply by j9k3r** [reply]
It looks like this is a joke of the new version of the browser, I have Chrome, what about you?

**Reply by j9k3r** [reply]
I was not entirely right, it’s all about Browsec VPN, you need to disable permission to use hardware acceleration in the system settings

**Reply by ZiadMohamedAlii** [reply]
MAAAN, I face the same problem did you solve it ????

proxyOverlay.js:1 Uncaught (in promise) TypeError: Cannot destructure property 'country' of '(intermediate value)' as it is null.

**Reply by Divyam19** [reply]
me too. I got the same problem in my next JS project one day it's working and the next it doesn't. just shows the error 404 page

**Reply by ZiadMohamedAlii** [reply]
How to solve it? I have tried ChatGPT but it didn't help

**Reply by teddythinh** [reply]
I got this error when setting up the axios call too.

**Reply by c9sid** [reply]
I am also getting the similer error in next.js "Error: Cannot destructure property 'topics' of '(intermediate value)' as it is undefined.

"

**Comment by webdevi121** [comment]
I solved it by changing my env from: WPGRAPHQL_URL     to:    GATSBY_WPGRAPHQL_URL.
What a pain, I just added the prefixed GATSBY_ in first place.
